pallets/messages/src/tests/mock.rs

```diff
@@ -68,25 +75,34 @@ impl system::Config for Test {
 	type OnSetCode = ();
 	type MaxConsumers = ConstU32<16>;
 	type SingleBlockMigrations = ();
-	type MultiBlockMigrator = ();
+	type MultiBlockMigrator = Migrator;
 	type PreInherents = ();
 	type PostInherents = ();
 	type PostTransactions = ();
 	type ExtensionsWeightInfo = ();
 }
 
+frame_support::parameter_types! {
+	pub storage MigratorServiceWeight: Weight = Weight::from_parts(100, 100); // do not use in prod
+}
+
+#[derive_impl(pallet_migrations::config_preludes::TestDefaultConfig)]
+impl pallet_migrations::Config for Test {
+	#[cfg(not(feature = "runtime-benchmarks"))]
+	type Migrations = (
+		crate::migration::v3::MigrateV2ToV3<Test, crate::SubstrateWeight<Test>>,
+		crate::migration::v3::FinalizeV3Migration<Test, crate::SubstrateWeight<Test>>,
+	);
+	#[cfg(feature = "runtime-benchmarks")]
+	type Migrations = pallet_migrations::mock_helpers::MockedMigrations;
+	type MaxServiceWeight = MigratorServiceWeight;
+}
+
 pub type MaxSchemaGrantsPerDelegation = ConstU32<30>;
 
 // Needs parameter_types! for the impls below
 parameter_types! {
-	// Max payload size was picked specifically to be large enough to accommodate
-	// a CIDv1 using SHA2-256, but too small to accommodate CIDv1 w/SHA2-512.
-	// This is purely so that we can test the error condition. Real world configuration
-	// should have this set large enough to accommodate the largest possible CID.
-	// Take care when adding new tests for on-chain (not IPFS) messages that the payload
-	// is not too big.
```

Comment:
This was an erroneous assumption. Since the configured value would never be this small on mainnet or testnet, it's better to allow a CIDv1/SHA2-512 by size, and have it fail for CID validation instead of for size reasons.